viewer/api/models.py

```diff
@@ -24,6 +24,12 @@ def __unicode__(self):
         return u'%s, date=%s, display=%s' % (
             self.name, self.date.strftime('%Y-%m-%d'), self.display)
 
+    def get_metrics(self):
+        metrics = set()
+        for collection in Collection.objects.filter(dataset=self):
+            metrics.add(collection.metric.id)
+        return list(metrics)
```

Comment:
Yes, good observation. I can go ahead and push a change there now. That will definitely help alleviate some backend processing of all those Collection objects.